protocols/autonatv2/src/client/handler/dial_back.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
+    convert::identity,
+    io,
+    task::{Context, Poll},
+};
+
+use either::Either;
+use futures::{AsyncRead, AsyncWrite, AsyncWriteExt};
+use futures_bounded::{FuturesSet, Timeout};
+use libp2p_core::upgrade::{DeniedUpgrade, ReadyUpgrade};
+use libp2p_swarm::{
+    handler::{ConnectionEvent, FullyNegotiatedInbound, ListenUpgradeError},
+    ConnectionHandler, ConnectionHandlerEvent, StreamProtocol, SubstreamProtocol,
+};
+
+use crate::request_response::DialBack;
+
+use super::DEFAULT_TIMEOUT;
+
+pub(crate) type ToBehaviour = Result<u64, io::Error>;
+
+pub(crate) struct Handler {
+    inbound: FuturesSet<io::Result<u64>>,
+}
+
+impl Handler {
+    pub(crate) fn new() -> Self {
+        Self {
+            inbound: FuturesSet::new(DEFAULT_TIMEOUT, 2),
```

Comment:
I am not sure if the constant here is actually beneficial. I'd rather have this inlined. Co-locating this with the actual functionality makes it much easier to judge whether the chosen values are appropriate.

Or if you don't want to have magic numbers, at least the constant should be in this file :)

Reply:
I changed that